### ISSUE TITLE
Explain routerScaleSize

### ISSUE THREAD

**Issue by Astagor** [edumeet/edumeet#295]
I have a question about the new parameter routerScaleSize.
From what I understand, it is designed to distribute peers using the same room over different mediaSoup workers. Right?
Why the default value of 20? In the post about the performance, it was written that there could be like 100 users in 1 room, and this was at a time when a room was bound to a single mediaSoup worker.

**Comment by havfo** [comment]
The reason for being conservative is that we want to support a worst case scenario, both in hardware and usage. 100 users works fine when you have a setting where a few people talk and a lot of people are stopping video and muting themselves. If you have a setting where everybody has video and audio the count goes down. You are correct though, 20 is probably a too low setting for most hardware, and we could reasonably assume something like 40-50.

To have an example, let us assume that (in a worst case) 10 people are unmuted at any given time in a meeting. Also let us assume that `lastN` for all participants in the call is on average 5. These assumptions are a bit high, but that is on purpose.

If we set `routerScaleSize` to 40, this would give us:
`(10 audio + 5 video) * routerScaleSize ~ 600 active consumers`

In other words, let us increase it to 40 by default, but newer hardware should have no trouble in having it higher. Also note that when all routers (cores) have been filled up to `routerScaleSize,` it continues to fill them up based on least load.

**Comment by misi** [comment]
Depends also on Last-N settings

**Comment by havfo** [comment]
Yes, in the example above I assumed `lastN ~ 5`. This will be your average of mobile users, normal users, and if someone has manually changed their settings.

**Comment by Astagor** [comment]
OK. So now it is more less a load balancer on the cores of the machin, not to overload 1 core with 1 room when there are a lot of users in this room. But then, is it not better to scale with core load from the beginning and not the number of peers?
So we assume when the core load is at 75%, then route peers to the next mediaSoup worker. If we route by number of peers like now, the cut off number varies depending on the hardware (faster or slower CPU). If we route based on the core load, it is then independent of the hardware.

**Comment by havfo** [comment]
I think we will have a hard time matching worker/router to CPU core to be able to check the load, and also, it would be very difficult to make an algorithm that can "allocate" a core for a router based on load. When you have a router you want to fill it up to some extent before moving to the next, the piping between routers is in no way free (from a CPU standpoint).

If we do in naively, this would potentially result in a room with 5 users on 5 different routers, and that is in no way optimal.

**Comment by Astagor** [comment]
OK, I understand.

**Comment by Astagor** [comment]
One more question, I thought that new room start always on a new router, but from TOP I see that for 7 rooms only 4 workers are loaded and 4 are idle (8 core machine). So you load a worker to the limit set in config, whatever the room is?

**Comment by havfo** [comment]
The routers are only shuffled, so random. That means that what you see can happen.
